apps/application-system/api/src/app/modules/application/template-api.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { ApplicationService } from './application.service'
+import { ApplicationWithAttachments } from '@island.is/application/core'
+import { BaseTemplateApiApplicationService } from '@island.is/application/template-api-modules'
+import { uuid } from 'uuidv4'
+import { AwsService } from './files'
+import {
+  ApplicationConfig,
+  APPLICATION_CONFIG,
+} from './application.configuration'
+
+@Injectable()
+export class TemplateApiApplicationService extends BaseTemplateApiApplicationService {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly awsService: AwsService,
+    @Inject(APPLICATION_CONFIG)
+    private readonly config: ApplicationConfig,
+  ) {
+    super()
+  }
+
+  async saveAttachmentToApplicaton(
+    application: ApplicationWithAttachments,
+    fileName: string,
+    buffer: Buffer,
+    uploadParameters?: {
+      ContentType?: string
+      ContentDisposition?: string
+      ContentEncoding?: string
+    },
+  ): Promise<string> {
+    const uploadBucket = this.config.attachmentBucket
+    if (!uploadBucket) throw new Error('No attachment bucket configured')
+
+    const fileId = uuid()
+    const key = `${application.id}/${fileId}-${fileName}`
```

Comment:
Super minor but currently the attachment key is `${fileId}-${fileName}` while the S3 key is `${application.id}/${fileId}-${fileName}`.

Would be good to keep this consistent.